cypress/component/category-select-table.cy.js

```diff
@@ -0,0 +1,116 @@
+import CategorySelectTable from "~/components/category-select-table.vue";
+
+// Documentation for testing Vue events of components
+// https://docs.cypress.io/guides/component-testing/events-vue
+
+const state = {
+
+    categories: [
+
+        "Subject ID",
+        "Age",
+        "Sex",
+        "Diagnosis",
+        "Assessment Tool"
+    ],
+
+    categoryClasses: {},
+
+    categoryToColorMap: {},
+
+    toolColorPalette: {
+
+        color1: "category-style-0",
+        color2: "category-style-1",
+        color3: "category-style-2",
+        color4: "category-style-3",
+        color5: "category-style-4",
+        colorDefault: "category-style-default"
+    },
+
+    getters: {
+
+        categories: () => state.categories,
+        categoryClasses: () => state.categoryClasses
```

Comment:
These two getters are the only thing that gets passed to the component. We should put the raw values that are going to be passed into the getters, so that we immediately know what data format the component expects to get from the getters. 

When I write a component test, I shouldn't have to worry about how to store manages to make the getters spit out the data this way, that's not my concern. And secondly, when I look at the test data, it should be easy to see what data will be fed in, I shouldn't have to reason about a factory function or see the test at runtime to get that information.

So concrete suggestion here:
- keep only `getters` in the `store` object, remove the rest
- explicitly set the `getters` to the values that'll get passed at runtime so I can see them. e.g.:

```javascript

    getters: {

        categories: () => [

            "Subject ID",
            "Age",
            "Sex",
            "Diagnosis",
            "Assessment Tool"
        ],
        categoryClasses: () => {
            return {
            "Subject ID": "category-style-0",
            "Age": "category-style-1",
            "Sex": "category-style-2",
            "Diagnosis": "category-style-3",
            "Assessment Tool": "category-style-4"
            };
        }
    }
```
- remove the function `setupCategoryClasses` and its invocation

If you like, I can send a patch or commit to your PR branch with the suggestion.

Reply:
Good comment. No, your meaning is entirely clear. In fact, for the `column-linking-table` test I had evolved my thinking to remove all such "factory" functions for store values. I will just remove that and make the values explicit as suggested.

Reply:
Resolved with https://github.com/neurobagel/annotation_tool/pull/229/commits/edea6979d53578997f8adecdded201ff65fed02c